src/ops/euclid.rs

```diff
@@ -0,0 +1,327 @@
+use core::ops::{Div, Rem};
+pub trait DivEuclid: Sized + Div<Self, Output = Self> {
+    /// Calculates Euclidean division, the matching method for `rem_euclid`.
+    ///
+    /// This computes the integer `n` such that
+    /// `self = n * v + self.rem_euclid(v)`.
+    /// In other words, the result is `self / v` rounded to the integer `n`
+    /// such that `self >= n * v`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use num_traits::DivEuclid;
+    ///
+    /// let a: i32 = 7;
+    /// let b: i32 = 4;
+    /// assert_eq!(DivEuclid::div_euclid(a,b), 1); // 7 > 4 * 1
```

Comment:
nit: this should be spaced `(a, b)` -- and there's a lot of similar non-standard formatting in places that `rustfmt` doesn't reach within docs and macros. Please go through and clean that up if you can.